sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -518,31 +533,93 @@ def refresh(self, **kwargs) -> None:
+                    logging.debug("Refresh all triggered by key: %s label %s.", key, label)
+                    return True, None
+            else:
+                raise e
+        return False, None
+
     def _load_all(self, **kwargs):
+        configuration_settings, sentinel_keys = self._load_configuration_settings(**kwargs)
+        if self._feature_flag_enabled:
+            feature_flags, feature_flag_sentinel_keys = self._load_feature_flags(**kwargs)
+            configuration_settings[FEATURE_MANAGEMENT_KEY] = {}
+            configuration_settings[FEATURE_MANAGEMENT_KEY][FEATURE_FLAG_KEY] = feature_flags
+            self._refresh_on_feature_flags = feature_flag_sentinel_keys
+        with self._update_lock:
+            self._refresh_on = sentinel_keys
+            self._dict = configuration_settings
+
+    def _load_configuration_settings(self, **kwargs):
         configuration_settings = {}
         sentinel_keys = kwargs.pop("sentinel_keys", self._refresh_on)
         for select in self._selects:
             configurations = self._client.list_configuration_settings(
                 key_filter=select.key_filter, label_filter=select.label_filter, **kwargs
             )
             for config in configurations:
-                key = self._process_key_name(config)
-                value = self._process_key_value(config)
                 if isinstance(config, FeatureFlagConfigurationSetting):
-                    feature_management = configuration_settings.get(FEATURE_MANAGEMENT_KEY, {})
-                    feature_management[key] = value
-                    if FEATURE_MANAGEMENT_KEY not in configuration_settings:
-                        configuration_settings[FEATURE_MANAGEMENT_KEY] = feature_management
+                    # Feature flags are ignored when loaded by Selects, as they are selected from
+                    # `feature_flag_selectors`
+                    pass
                 else:
+                    key = self._process_key_name(config)
+                    value = self._process_key_value(config)
                     configuration_settings[key] = value
                 # Every time we run load_all, we should update the etag of our refresh sentinels
                 # so they stay up-to-date.
                 # Sentinel keys will have unprocessed key names, so we need to use the original key.
                 if (config.key, config.label) in self._refresh_on:
                     sentinel_keys[(config.key, config.label)] = config.etag
-        self._refresh_on = sentinel_keys
-        with self._update_lock:
-            self._dict = configuration_settings
+        return configuration_settings, sentinel_keys
+
+    def _load_feature_flags(self, **kwargs):
+        feature_flag_sentinel_keys = {}
+        loaded_feature_flags = []
+        # Needs to be removed unknown keyword argument for list_configuration_settings
```

Comment:
> `unknown keyword argument` 

Is that an actual error that will be thrown?

Reply:
The SDK cares, we can't pass it any unknown kwargs or they thrown an error. My understanding is they check if there are any un popped kwargs just before the request and throw an error if there is any. I've talked to them about it. They don't like it but it would take a lot of rework to fix.